src/parsers/manifest/dash/node_parsers/index.ts

```diff
@@ -397,250 +337,352 @@ export default function parseManifest(
+          const parsedAdaptation = parsedAdaptations[type];
+          if (parsedAdaptation) {
+            parsedAdaptation.push(parsedAdaptationSet);
+          } else {
+            parsedAdaptations[type] = [parsedAdaptationSet];
           }
-
-          return parsedRepresentation;
-        });
-
-      const adaptationMimeType = adaptation.attributes.mimeType;
-      const adaptationCodecs = adaptation.attributes.codecs;
-
-      const representationMimeTypes = representations
-        .map(r => r.mimeType)
-        .filter((mimeType : string|undefined) : mimeType is string => mimeType != null);
-
-      const representationCodecs = representations
-        .map(r => r.codecs)
-        .filter((codecs : string|undefined) : codecs is string => codecs != null);
-
-      const type = inferAdaptationType(
-        adaptationMimeType || null,
-        representationMimeTypes,
-        adaptationCodecs || null,
-        representationCodecs,
-        adaptationChildren.role || null
-      );
-
-      let closedCaption : boolean|undefined;
-      let audioDescription : boolean|undefined;
-
-      if (
-        type === "text" &&
-        adaptationChildren.accessibility &&
-        isHardOfHearing(adaptationChildren.accessibility)
-      ) {
-        closedCaption = true;
-      }
-
-      if (
-        type === "audio" &&
-        adaptationChildren.accessibility &&
-        isVisuallyImpaired(adaptationChildren.accessibility)
-      ) {
-        audioDescription = true;
-      }
-
-      let adaptationID : string;
-      if (adaptation.attributes.id != null) {
-        adaptationID = adaptation.attributes.id;
-      } else {
-        let idString = type;
-        if (adaptation.attributes.language) {
-          idString += `-${adaptation.attributes.language}`;
-        }
-        if (closedCaption) {
-          idString += "-cc";
-        }
-        if (audioDescription) {
-          idString += "-ad";
-        }
-        if (adaptation.attributes.contentType) {
-          idString += `-${adaptation.attributes.contentType}`;
-        }
-        if (adaptation.attributes.codecs) {
-          idString += `-${adaptation.attributes.codecs}`;
         }
-        if (adaptation.attributes.mimeType) {
-          idString += `-${adaptation.attributes.mimeType}`;
-        }
-        if (adaptation.attributes.frameRate) {
-          idString += `-${adaptation.attributes.frameRate}`;
-        }
-
-        if (idString.length === type.length) {
-          idString += representations.length ?
-            ("-" + representations[0].id) : "-empty";
-        }
-
-        adaptationID = "adaptation-" + idString;
-      }
-
-      const parsedAdaptationSet : IParsedAdaptation = {
-        id: adaptationID,
-        representations,
-        type,
+        return parsedAdaptations;
+      }, {});
+
+      const parsedPeriod : IParsedPeriod = {
+        id: periodID,
+        start: periodStart,
+        duration: periodDuration,
+        adaptations,
```

Comment:
Where do you put the main adaptation first?

Reply:
Done